Jetcaster/wear/src/main/java/com/example/jetcaster/ui/library/PodcastsScreen.kt

```diff
@@ -0,0 +1,202 @@
+ */
+
+package com.example.jetcaster.ui.library
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.wear.compose.foundation.lazy.rememberScalingLazyListState
+import androidx.wear.compose.material.ButtonDefaults
+import androidx.wear.compose.material.ChipDefaults
+import androidx.wear.compose.material.MaterialTheme
+import androidx.wear.compose.material.Text
+import androidx.wear.compose.material.dialog.Alert
+import androidx.wear.compose.material.dialog.Dialog
+import com.example.jetcaster.R
+import com.example.jetcaster.core.data.model.PodcastInfo
+import com.google.android.horologist.annotations.ExperimentalHorologistApi
+import com.google.android.horologist.composables.PlaceholderChip
+import com.google.android.horologist.composables.Section
+import com.google.android.horologist.composables.SectionedList
+import com.google.android.horologist.compose.layout.ScreenScaffold
+import com.google.android.horologist.compose.layout.rememberColumnState
+import com.google.android.horologist.compose.material.Button
+import com.google.android.horologist.compose.material.Chip
+import com.google.android.horologist.compose.material.Title
+import com.google.android.horologist.images.coil.CoilPaintable
+
+@Composable
+fun PodcastsScreen(
+    podcastViewModel: PodcastViewModel = hiltViewModel(),
+    onPodcastsItemClick: (PodcastInfo) -> Unit,
+    onErrorDialogCancelClick: () -> Unit,
+) {
+    val uiState by podcastViewModel.uiState.collectAsStateWithLifecycle()
+
+    val modifiedState = when (uiState) {
+        is PodcastsScreenState.Loaded -> {
+            val modifiedPodcast = (uiState as PodcastsScreenState.Loaded).podcastList.map {
+                it.takeIf { it.title.isNotEmpty() }
+                    ?: it.copy(title = stringResource(id = R.string.no_title))
+            }
+
+            PodcastsScreenState.Loaded(modifiedPodcast)
+        }
+
+        PodcastsScreenState.Failed,
+        PodcastsScreenState.Loading,
+        -> uiState
+    }
+
+    PodcastsScreen(
+        podcastsScreenState = modifiedState,
+        onPodcastsItemClick = onPodcastsItemClick
+    )
+
+    if (modifiedState == PodcastsScreenState.Failed) {
+        Dialog(
+            showDialog = true,
```

Comment:
This can be simplified as:
```suggestion
            showDialog = modifiedState == PodcastsScreenState.Failed,
```

...and remove the outer if clause

Reply:
thanks